internal/provider/resource_access_generic_test.go

```diff
@@ -176,3 +180,122 @@ func testAccCheckJaasResourceAccess(relation string, object, target *string, exp
 		return nil
 	}
 }
+
+func TestDiffStringSets(t *testing.T) {
+	ctx := context.Background()
+
+	testCases := []struct {
+		desc     string
+		current  []string
+		target   []string
+		expected []string
+	}{
+		{
+			desc:     "Empty sets",
+			current:  []string{},
+			target:   []string{},
+			expected: []string{},
+		},
+		{
+			desc:     "Identical sets",
+			current:  []string{"a", "b"},
+			target:   []string{"a", "b"},
+			expected: []string{},
+		},
+		{
+			desc:     "Disjoint sets",
+			current:  []string{"a", "b"},
+			target:   []string{"c", "d"},
+			expected: []string{"a", "b"},
+		},
+		{
+			desc:     "Overlapping sets",
+			current:  []string{"a", "b", "c"},
+			target:   []string{"b", "c", "d"},
+			expected: []string{"a"},
+		},
+		{
+			desc:     "Subset",
+			current:  []string{"a", "b"},
+			target:   []string{"a", "b", "c"},
+			expected: []string{},
+		},
+		{
+			desc:     "Superset",
+			current:  []string{"a", "b", "c"},
+			target:   []string{"a", "b"},
+			expected: []string{"c"},
+		},
+	}
+
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			currentSet, _ := basetypes.NewSetValueFrom(ctx, types.StringType, tC.current)
+			targetSet, _ := basetypes.NewSetValueFrom(ctx, types.StringType, tC.target)
+			var diags diag.Diagnostics
+
+			result := diffStringSets(currentSet, targetSet, &diags)
+
+			assert.False(t, diags.HasError())
+
+			var resultSlice []string
+			result.ElementsAs(ctx, &resultSlice, false)
+			assert.ElementsMatch(t, tC.expected, resultSlice)
+		})
+	}
+}
+
+func TestDiffStringSets_Validation(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("Invalid current set type", func(t *testing.T) {
```

Comment:
This style of tests was done by the AI but it turns out to be very handy, in one larger test you can see where a subtest starts/ends and you can even run the subtest directly from your IDE.